src/Microsoft.OpenApi/Models/OpenApiSchema.cs

```diff
@@ -557,6 +553,10 @@ public void SerializeInternal(IOpenApiWriter writer, OpenApiSpecVersion version,
             writer.WriteProperty(OpenApiConstants.Deprecated, Deprecated, false);
 
             // extensions
+            if (Extensions.ContainsKey(OpenApiConstants.NullableExtension))
```

Comment:
I do not believe we should making any changes to the document when writing.  If a developer chooses to add an x-Nullable extension in a 3.1 version document, then so be it. We should write it out as they requested.

Reply:
We should however handle removing the x-Nullable and Nullable during reading.